### ISSUE TITLE
PostgreSQL interval parser handle fractional seconds with interval style iso_8601

### ISSUE THREAD

**Issue by rocketraman** [jOOQ/jOOQ#14699]
### Expected behavior

I have a PostgresQL database with `intervalstyle` set to `iso_8601` (see https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-INTERVAL-OUTPUT).

The parser for this type does not handle fractional seconds -- it assumes all components of the time value are integers:

https://github.com/jOOQ/jOOQ/blob/5ebcc5c0531e3eeee95ce809ef9602c5a2e0974e/jOOQ/src/main/java/org/jooq/util/postgres/PGInterval.java#L98

An example input to this function is `PT6M32.417S`.

### Actual behavior

```
Exception in thread "main" org.jooq.exception.DataAccessException: SQL[...]; Error while reading field: "public"."table"."field", at JDBC index: 83
	at org.jooq_3.17.6.POSTGRES.debug(Unknown Source)
	at org.jooq_3.17.6.POSTGRES.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3313)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:678)
	at org.jooq.impl.CursorImpl$CursorIterator.fetchNext(CursorImpl.java:1464)
	at org.jooq.impl.CursorImpl$CursorIterator.hasNext(CursorImpl.java:1425)
	at org.jooq.impl.CursorImpl.fetchNext(CursorImpl.java:238)
	at org.jooq.impl.AbstractCursor.fetch(AbstractCursor.java:177)
	at org.jooq.impl.AbstractCursor.fetch(AbstractCursor.java:88)
	at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:265)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:341)
	at org.jooq.impl.AbstractResultQuery.fetch(AbstractResultQuery.java:290)
	at org.jooq.impl.SelectImpl.fetch(SelectImpl.java:2837)
	at org.jooq.impl.Tools$3$1.block(Tools.java:5854)
	at java.base/java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3463)
	at java.base/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3434)
	at org.jooq.impl.Tools$3.get(Tools.java:5851)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run$$$capture(CompletableFuture.java:1768)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java)
	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
Caused by: java.sql.SQLException: Error while reading field: "public"."event_user_checkins"."delta", at JDBC index: 83
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1590)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.apply(CursorImpl.java:1525)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.apply(CursorImpl.java:1484)
	at org.jooq.impl.RecordDelegate.operate(RecordDelegate.java:144)
	at org.jooq.impl.CursorImpl$CursorIterator.fetchNext(CursorImpl.java:1449)
	... 20 more
Caused by: java.lang.NumberFormatException: For input string: "32.417"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Integer.parseInt(Integer.java:668)
	at java.base/java.lang.Integer.parseInt(Integer.java:786)
	at org.jooq.util.postgres.PGInterval.parseISO8601Format(PGInterval.java:98)
	at org.jooq.util.postgres.PGInterval.setValue(PGInterval.java:144)
	at org.jooq.util.postgres.PGInterval.<init>(PGInterval.java:49)
	at org.jooq.util.postgres.PostgresUtils.toDayToSecond(PostgresUtils.java:250)
	at org.jooq.impl.DefaultBinding$DefaultDayToSecondBinding.get0(DefaultBinding.java:2464)
	at org.jooq.impl.DefaultBinding$DefaultDayToSecondBinding.get0(DefaultBinding.java:2428)
	at org.jooq.impl.DefaultBinding$InternalBinding.get(DefaultBinding.java:1024)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1580)
	... 24 more
```

### Steps to reproduce the problem

```
alter database dbname set INTERVALSTYLE TO "iso_8601"

INSERT INTO public.intervaltable (intervalcolumn) VALUES ('0 years 0 mons 0 days 1 hours 31 mins 17.883 secs');
```

Then use Jooq to select `intervalcolumn` from `intervaltable`.

### jOOQ Version

3.17.6

### Database product and version

PostgreSQL 12.11 on x86_64-pc-linux-gnu, compiled by Debian clang version 12.0.1, 64-bit

### Java Version

java version "17.0.5" 2022-10-18 LTS

### OS Version

Fedora Linux 37 (Thirty Seven)

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.2.27.jre7

**Comment by lukaseder** [comment]
Thanks a lot for your report. That particular class has been copied and shaded from pgjdbc (see the copyright header). This is to avoid adding a third party dependency only for this: https://github.com/jOOQ/jOOQ/issues/13053

May I suggest offering this patch upstream to https://github.com/pgjdbc/pgjdbc ? We'll reproduce whatever the pgjdbc team will accept as a fix, later.

**Comment by lukaseder** [comment]
I'll keep this issue open to make sure we reflect any pgjdbc changes to this class, as soon as they're available.

**Comment by lukaseder** [comment]
Note, if you create an issue / PR at https://github.com/pgjdbc/pgjdbc, can you please link it from here?

**Comment by rocketraman** [comment]
@lukaseder It looks like the upstream version of this class already has this bug fixed. See https://github.com/pgjdbc/pgjdbc/blob/99d010147b3e6907a39485ec0113c56d09fdfc37/pgjdbc/src/main/java/org/postgresql/util/PGInterval.java#L106.

It was fixed in https://github.com/pgjdbc/pgjdbc/commit/73bac99f8b2b192c3c76fa160616bb6c64bc93e9 only a week ago.

**Comment by lukaseder** [comment]
Ah yes, I see, thanks, the issue referenced is this one: https://github.com/pgjdbc/pgjdbc/pull/2457

And it's been released already. I'll look into this right away.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.9 (https://github.com/jOOQ/jOOQ/issues/14709), and 3.16.15 (https://github.com/jOOQ/jOOQ/issues/14710).

Thanks again for your report and help!

**Comment by lukaseder** [comment]
The fix cannot be backported to 3.16.15, because of #13053 being available only in jOOQ 3.17+
